File: Flix.Backend/Flix.Services/Database/UserReport.cs
using Flix.Services.Enums;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text;

namespace Flix.Services.Database
{
    public class UserReport
    {
        [Key]
        public int Id { get; set; }
       
        public int ReporterId { get; set; }
        [ForeignKey(nameof(ReporterId))]
        public User Reporter { get; set; } = null!;

        public int ReportedUserId { get; set; }
        [ForeignKey(nameof(ReportedUserId))]
        public User ReportedUser { get; set; } = null!;

        [Required]
        [MaxLength(1000)]
        public string Reason { get; set; } = string.Empty;

        public ReportStatus Status { get; set; } = ReportStatus.Open;

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public int? ReviewedByUserId { get; set; }
        [ForeignKey(nameof(ReviewedByUserId))]
        public User? ReviewedBy { get; set; }
        public DateTime? ResolvedAt { get; set; }
        public string? AdminComment { get; set; }
    }
}

File: Flix.UI/flix_mobile/lib/models/user_report.dart
import 'package:flix_mobile/enums/report_status.dart';
import 'package:flix_mobile/models/user.dart';
import 'package:json_annotation/json_annotation.dart';

part 'user_report.g.dart';

@JsonSerializable()
class UserReport {
  UserReport(
    this.id,
    this.reportedUserId,
    this.header,
    this.description,
    this.status,
    this.createdAt,
    this.reporter,
    this.reportedUser,
    this.reviewedBy,
    this.resolvedAt,
    this.adminComment,
  );

  final int? id;
  final int? reportedUserId;

  final String? header;
  final String? description;

  final ReportStatus? status;
  final DateTime? createdAt;

  final User? reporter;
  final User? reportedUser;
  final User? reviewedBy;

  final DateTime? resolvedAt;
  final String? adminComment;

  String? get statusLabel => status == null ? null : getReportStatus(status!);

  factory UserReport.fromJson(Map<String, dynamic> json) =>
      _$UserReportFromJson(json);

  Map<String, dynamic> toJson() => _$UserReportToJson(this);
}
